upstairs/src/volume.rs

```diff
@@ -1536,52 +1558,50 @@ impl Volume {
                     )
                 }
 
-                // Presently, we only support one sub_volume for replacement.
-                // If support for multiple sub_volumes is added, then this
-                // following section will need to be updated to loop over the
-                // sub_volume Vec and find the specific one with a difference,
-                // while verifying that all other sub_volumes are no different.
-                if n_sub_volumes.len() != 1 {
-                    crucible_bail!(
-                        ReplaceRequestInvalid,
-                        "Only a single sub_volume is supported"
-                    )
+                let mut sv_results = Vec::new();
+                // Walk the list of SubVolumes, get the compare result from
+                // each one and add it to list for the final result.
+                // If we get an error, then that is returned right away and
+                // we don't bother processing any further.
+                for (o_sv, new_sv) in
+                    o_sub_volumes.iter().zip(n_sub_volumes.iter())
+                {
+                    let sv_res = Self::compare_vcr_for_replacement(
+                        log, o_sv, new_sv, read_only,
+                    )?;
+                    sv_results.push(sv_res);
                 }
+                let read_only_parent_compare =
+                    Box::new(match (o_read_only_parent, n_read_only_parent) {
+                        (.., None) => {
+                            // The New VCR is none. When comparing
```

Comment:
"The New VCR is none" is confusing.  I think it means "The new VCR has no read-only parent"?  Also, saying _why_ this is okay would be helpful (I think we need to support the case because it occurs after scrubbing is complete?)

Reply:
Updated the comment in ab9e0223